framework/src/headers/ip/v6/icmp/mod.rs

```diff
@@ -117,37 +125,47 @@ where
     }
 }
 
-impl<T> IcmpV6Header<T>
+impl<T> Icmpv6Header<T>
 where
     T: IpHeader,
 {
+    #[inline]
+    pub fn new() -> Self {
+        Default::default()
+    }
+
     #[inline]
     pub fn msg_type(&self) -> Option<IcmpMessageType> {
-        FromPrimitive::from_u8(u8::from_be(self.msg_type))
+        FromPrimitive::from_u8(self.msg_type)
```

Comment:
so when dealing w/ u8, the conversions are unnecessary for LE/BE because it's the same either way. As per https://github.com/rust-lang/rust/issues/44724, the functions are only included for generality and macro generation.